File: clients/python/model/server_message_game.py
from .player_view import PlayerView
class ServerMessageGame:
    def __init__(self, player_view):
        self.player_view = player_view
    @staticmethod
    def read_from(stream):
        if stream.read_bool():
            player_view = PlayerView.read_from(stream)
        else:
            player_view = None
        return ServerMessageGame(player_view)
    def write_to(self, stream):
        if self.player_view is None:
            stream.write_bool(False)
        else:
            stream.write_bool(True)
            self.player_view.write_to(stream)
    def __repr__(self):
        return "ServerMessageGame(" + \
            repr(self.player_view) + \
            ")"

File: clients/javascript/model/server-message-game.js
const PlayerView = require('./player-view').PlayerView;
class ServerMessageGame {
    constructor(playerView) {
        this.playerView = playerView;
    }
    static async readFrom(stream) {
        let playerView;
        if (await stream.readBool()) {
            playerView = await PlayerView.readFrom(stream);
        } else {
            playerView = null;
        }
        return new ServerMessageGame(playerView);
    }
    async writeTo(stream) {
        let playerView = this.playerView;
        if (playerView === null) {
            await stream.writeBool(false);
        } else {
            await stream.writeBool(true);
            await playerView.writeTo(stream);
        }
    }
}
module.exports = { ServerMessageGame: ServerMessageGame }
